src/RestApi/StoreApi/Utilities/MoneyValue.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Handles monetary values using integers by using the smallest unit of a currency.
+ *
+ * This could be replaced with https://github.com/moneyphp/money if we want to
+ * support currencies in the future. However at this stage we only need to use
+ * the store decimal setting (which is independent of currency) to format
+ * values for the API.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * MoneyValue class.
+ */
+class MoneyValue {
+
+	/**
+	 * Monetary amount. Integer stored as string.
+	 *
+	 * @var string
+	 */
+	protected $amount;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string|integer $amount Set money amount.
+	 */
+	public function __construct( $amount = '0' ) {
+		$this->amount = (string) $amount;
+	}
+
+	/**
+	 * Converts to string if this class is used as a string.
+	 */
+	public function __toString() {
+		return $this->amount;
+	}
+
+	/**
+	 * Return the price amount.
+	 *
+	 * @return string|int
+	 */
+	public function get_amount() {
+		return $this->amount > PHP_INT_MAX ? $this->amount : (int) $this->amount;
```

Comment:
I wonder if we should avoid mixed return types and just stick to a specific type (i.e. only return string)?  We could maybe include a method that allows php code to know that it shouldn't be converted back to integer (`exceeds_php_max_int`) or something like that.

Reply:
I only added this to be safe... Using integers to me makes the most sense for both this and the responses, but being mindful of PHP_INT_MAX where strings would be needed so the responses due not break for those large integer edge cases. Would documentation here make the most sense or do we want to be strict and only use string?

Reply:
If we return one of two types, that means consumers always need to test for those two types, whereas if we always return string, consumer knows it always needs to convert it, and can convert depending on what expectations it may have for the value (whether it potentially exceeds the max integer for the language in use - which has more relevance over the REST api than internally in php).  For instance the value of `PHP_INT_MAX` is `2,147,483,647`, whereas the value for javascripts Number.MAX_SAFE_INTEGER is `9007199254740991` (wildly different numbers).  So javascript can pretty much reliably just convert the REST Api response to a number and be done with it (highly unlikely a money value will go into the quadrillions)... where-as php side - it might depend on what currency is in use and how many subunits it has...